### PR TITLE
Remove connection type from endpoint network rules

### LINKED ISSUE #419: [Rule Tuning] Endpoint rules using network data and event.type:connection
**tl;dr**: any rule using the index `logs-endpoint.events.*` (or `winlogbeat-*` and _can_ be compatible with endpoint) should not be using `event.type:connection` for network events.

## Description
There are many endpoint rules which are using `event.type:connection` for network data (`event.category:network`). The endpoint sensor does not populate `connection` to that field and so network rules would not be compatible across `winlogbeat-*` and `logs-endpoint.events.*` indexes.

Observed `network.protocol` correlations for 7.10 agents (per my limited testing dataset):
`event.type` | Endpoint | winlogbeat
-------------|----------|-----------
start | _missing_ | -, https, netbios-ns, llmnr, http, ssdp, domain, bootps, dhcpv6-server, ntp
end | _missing_ | _missing_
protocol | dns, http | -, https, netbios-ns, llmnr, http, ssdp, domain, bootps, dhcpv6-server, ntp, dns
info | dns | dns

`network.protocol` fields for `event.type:connection` in winlogbeat match up to `the protocol` fields, with both fields populated for backwards compatibility (it is a multi-value field)

Ideally both datasources will achieve parity in data and mappings, but until then, all impacted rules will need to have `connection` removed and have the `event.type` updated and logic adapted accordingly. We also should ensure ECS is being applied consistently/correctly across the datasources using these fields.

### PR DESCRIPTION
**Pull request by brokensound77**
## Issues
resolves #419

## Summary
Modified all rules using the index `logs-endpoint.events.*` (or `winlogbeat-*` which can be compatible with endpoint) that were using `event.type:connection` for network events.

For just about all of these modified rules, the use of `event.type` was used to limit the number of alerts by restricting to only a subset of network events. Even with the `event.type` defined, the rules would still trigger many alerts. The best way to limit this is to use an `eql` sequence off the process `start` event, which would limit it to a single alert.

Also, see the issue for inconsistencies across the two data sources.

### CODE REVIEW

File: rules/linux/lateral_movement_telnet_network_activity_external.toml [hunk 1]
```diff
@@ -20,20 +20,21 @@ false_positives = [
...
 timeline_id = "76e52245-7519-4251-91ab-262fb1a1728c"
-type = "query"
+type = "eql"
 
 query = '''
-event.category:network and event.type:(connection or start) and
-  process.name:telnet and
-  not destination.ip:(127.0.0.0/8 or 10.0.0.0/8 or 172.16.0.0/12 or 192.168.0.0/16 or "FE80::/10" or "::1/128")
+sequence by process.entity_id
+  [process where process.name : "telnet" and event.type == "start"]
```
Comment (brokensound77) [inline rules/linux/lateral_movement_telnet_network_activity_external.toml:35]: `process.name : "telnet"` could possibly be changed to `process.name == "telnet"` because Linux

File: rules/windows/defense_evasion_rundll32_no_arguments.toml [hunk 2]
```diff
@@ -29,9 +29,7 @@ sequence with maxspan=1h
      (process.name : "rundll32.exe" /* or process.pe.original_file_name == "RUNDLL32.EXE" */ ) and
      process.args_count < 2
   ] by process.entity_id
-  [process where event.type in ("start", "process_started") and
-                                    /* uncomment once in winlogbeat */
-     (process.name : "rundll32.exe" /* or process.pe.original_file_name == "RUNDLL32.EXE" */ )
+  [process where event.type in ("start", "process_started") and process.parent.name : "rundll32.exe"
```
Comment (brokensound77) [inline rules/windows/defense_evasion_rundll32_no_arguments.toml:32]: this is an unrelated correction to faulty logic which @paulewing found

File: rules/linux/lateral_movement_telnet_network_activity_external.toml (hunk 1 above)
Comment (rw-access) [reply rules/linux/lateral_movement_telnet_network_activity_external.toml:35]: definitely! i think linux should use `==` when possible

File: rules/linux/lateral_movement_telnet_network_activity_external.toml [hunk 3]
```diff
@@ -20,20 +20,21 @@ false_positives = [
...
+  [network where process.name : "telnet" and
+    destination.ip not in ("127.0.0.0/8", "10.0.0.0/8", "172.16.0.0/12", "192.168.0.0/16", "FE80::/10", "::1/128")]
```
Comment (rw-access) [inline rules/linux/lateral_movement_telnet_network_activity_external.toml:37]:
```suggestion
    not cidrMatch("127.0.0.0/8", "10.0.0.0/8", "172.16.0.0/12", "192.168.0.0/16", "FE80::/10", "::1/128")]
```

File: rules/linux/lateral_movement_telnet_network_activity_internal.toml [hunk 4]
```diff
@@ -20,21 +20,22 @@ false_positives = [
...
 
 query = '''
-event.category:network and event.type:(connection or start) and
-  process.name:telnet and
-  destination.ip:((10.0.0.0/8 or 172.16.0.0/12 or 192.168.0.0/16 or "FE80::/10") and
-                    not (127.0.0.0/8 or "::1/128"))
+sequence by process.entity_id
+  [process where process.name : "telnet" and event.type == "start"]
+  [network where process.name : "telnet" and
+    destination.ip in ("10.0.0.0/8", "172.16.0.0/12", "192.168.0.0/16", "FE80::/10") and
```
Comment (rw-access) [inline rules/linux/lateral_movement_telnet_network_activity_internal.toml:37]: cidrMatch

File: rules/linux/linux_netcat_network_connection.toml [hunk 5]
```diff
@@ -33,10 +33,14 @@ rule_id = "adb961e0-cb74-42a0-af9e-29fc41f88f5f"
...
 tags = ["Elastic", "Host", "Linux", "Threat Detection"]
 timeline_id = "76e52245-7519-4251-91ab-262fb1a1728c"
-type = "query"
+type = "eql"
 
 query = '''
-event.category:network and event.type:(access or connection or start) and
-  process.name:(nc or ncat or netcat or netcat.openbsd or netcat.traditional)
+sequence by process.entity_id
+  [process where (process.name : "nc" or process.name : "ncat" or process.name : "netcat" or
```
Comment (rw-access) [inline rules/linux/linux_netcat_network_connection.toml:40]: these `or`s make me sad. but i get it

File: rules/windows/execution_command_prompt_connecting_to_the_internet.toml [hunk 6]
```diff
@@ -18,20 +18,21 @@ false_positives = [
...
+type = "eql"
 
 query = '''
-event.category:network and event.type:connection and
-  process.name:cmd.exe and
-  not destination.ip:(10.0.0.0/8 or 172.16.0.0/12 or 192.168.0.0/16)
+sequence by process.entity_id
+  [process where process.name : "cmd.exe" and event.type == "start"]
+  [network where process.name : "cmd.exe" and
+     destination.ip not in ("10.0.0.0/8", "172.16.0.0/12", "192.168.0.0/16")]
```
Comment (rw-access) [inline rules/windows/execution_command_prompt_connecting_to_the_internet.toml:35]: cidrMatch?

File: rules/linux/lateral_movement_telnet_network_activity_external.toml (hunk 3 above)
Comment (rw-access) [reply rules/linux/lateral_movement_telnet_network_activity_external.toml:37]: actually... we might not need to worry about this

**Review by rw-access — APPROVED** [review]
i'm on the fence about cidrmatch but otherwise lgtm

**Review by bm11100 — APPROVED** [review]
looks good and should be easy enough to test these

File: rules/linux/lateral_movement_telnet_network_activity_external.toml (hunk 1 above)
Comment (brokensound77) [reply rules/linux/lateral_movement_telnet_network_activity_external.toml:35]:
```suggestion
  [process where process.name == "telnet" and event.type == "start"]
```

File: rules/linux/lateral_movement_telnet_network_activity_external.toml [hunk 7]
```diff
@@ -20,20 +20,21 @@ false_positives = [
...
-type = "query"
+type = "eql"
 
 query = '''
-event.category:network and event.type:(connection or start) and
-  process.name:telnet and
-  not destination.ip:(127.0.0.0/8 or 10.0.0.0/8 or 172.16.0.0/12 or 192.168.0.0/16 or "FE80::/10" or "::1/128")
+sequence by process.entity_id
+  [process where process.name : "telnet" and event.type == "start"]
+  [network where process.name : "telnet" and
```
Comment (brokensound77) [inline rules/linux/lateral_movement_telnet_network_activity_external.toml:36]:
```suggestion
  [network where process.name == "telnet" and
```

File: rules/linux/lateral_movement_telnet_network_activity_internal.toml [hunk 8]
```diff
@@ -20,21 +20,22 @@ false_positives = [
...
+type = "eql"
 
 query = '''
-event.category:network and event.type:(connection or start) and
-  process.name:telnet and
-  destination.ip:((10.0.0.0/8 or 172.16.0.0/12 or 192.168.0.0/16 or "FE80::/10") and
-                    not (127.0.0.0/8 or "::1/128"))
+sequence by process.entity_id
+  [process where process.name : "telnet" and event.type == "start"]
+  [network where process.name : "telnet" and
```
Comment (brokensound77) [inline rules/linux/lateral_movement_telnet_network_activity_internal.toml:36]:
```suggestion
  [process where process.name == "telnet" and event.type == "start"]
  [network where process.name == "telnet" and
```

File: rules/linux/linux_netcat_network_connection.toml [hunk 9]
```diff
@@ -33,10 +33,14 @@ rule_id = "adb961e0-cb74-42a0-af9e-29fc41f88f5f"
...
+                  process.name : "netcat.openbsd" or process.name : "netcat.traditional") and
+     event.type == "start"]
+  [network where (process.name : "nc" or process.name : "ncat" or process.name : "netcat" or
+                  process.name : "netcat.openbsd" or process.name : "netcat.traditional")]
```
Comment (brokensound77) [inline rules/linux/linux_netcat_network_connection.toml:44]:
```suggestion
  [process where (process.name == "nc" or process.name == "ncat" or process.name == "netcat" or
                  process.name == "netcat.openbsd" or process.name == "netcat.traditional") and
     event.type == "start"]
  [network where (process.name == "nc" or process.name == "ncat" or process.name == "netcat" or
                  process.name == "netcat.openbsd" or process.name == "netcat.traditional")]
```
